spyder/__init__.py

```diff
@@ -93,8 +93,13 @@ def get_versions(reporev=True):
         'system': platform.system(),   # Linux, Windows, ...
         'release': platform.release(),  # XP, 10.6, 2.2.0, etc.
         'revision': revision,  # '9fdf926eccce',
-        'branch': branch,  # '4.x' or master
+        'branch': branch,  # '4.x' or master,
+        'machine': platform.machine(),  # 'arm64', 'x86_64', 'AMD64', ...
     }
+    if sys.platform == 'darwin':
+        versions.update(system='macOS', release=platform.mac_ver()[0])
```

Comment:
Instead of this manually and somewhat inelegant hardcoding, what about using the built in [`platform.system_alias`](https://docs.python.org/3.12/library/platform.html?highlight=platform#platform.system_alias) instead which is purpose-built for this (and can handle more than just macOS)?

Reply:
What will be gained by using `platform.system_alias`? Hardcoding is still required as I must provide `'macOS'`, `platform.mac_ver()[0]`, and `platform.version()` as positional arguments to `platform.system_alias`. As far as I can tell, `platform.system_alias` is utterly and completely useless. But if I'm missing something, please enlighten me.

Reply:
Investigating `platform.system_alias` source code reveals that macOS is explicitly not handled.

Reply:
> Hardcoding is still required as I must provide `'macOS'`, `platform.mac_ver()[0]`, and `platform.version()` as positional arguments to `platform.system_alias`.

The idea was that you would pass `platform.system()`, `platform.release()` and `platform.version()` but it seems as you mention that doesn't actually work for macOS, and is also rather verbose.

A potentially much more sensible approach that should actually work, though and is much more concise is adding a new key `platform` with the output of [`platform.platform(aliased=True)`](https://docs.python.org/3.12/library/platform.html?highlight=platform#platform.platform), which contains the desired information in human-readible form (platform, release and version), and use that instead. As of Python 3.8, it is documented to return macOS and `mac_ver` rather than the Darwin version, thus avoiding the need for this replacement.

> Investigating `platform.system_alias` source code reveals that macOS is explicitly not handled.

Oh, oops—I assumed it would handle that, but apparently not per python/cpython#11207

Reply:
> `platform.platform(aliased=True)`

These are the results that I get...
* macOS: macOS-13.3-x86_64-i386-64bit
* Ubuntu: Linux-5.19.0-38-generic-x86_64-with-glibc2.36
* Windows: Windows-10-10.0.19045-SP0

It seems this may be suitable for issue reporting. Windows doesn't display machine, 'AMD64', although this is probably not as critical as it is for macOS and Linux.
For the About widget, parsing this correctly for Linux would be problematic, so it may be best to leave that as is, except for macOS.

So, I would summarize your proposal as:
* Set a new key `'platform'` with value from `platform.platform`
* Use `'platform'` key for issue reporting, since pretty formatting is not necessary
* For macOS only, the system info for the About widget can be parsed from `'platform'` value to avoid hard-coding

Is that a faithful summary? If so, I don't have any issues with it.

Reply:
Yup, aside from a couple small details—

>  For the About widget, parsing this correctly for Linux would be problematic

Yeah, it is intended to be human-readable and output as-is and is not supposed to be parsed (that's what `platform.system()`, etc are for), so it shouldn't be used there.

> For macOS only, the system info for the About widget can be parsed from 'platform' value to avoid hard-coding

IMO, that isn't worth the complexity as we're parsing a human-readible string rather than something actually intended to be parsed, and thus would require more assumptions and hardcoding than replacing `darwin` for `macOS`. Therefore, I'd suggest just doing the latter if nessesary for the about dialog.